runtime/service.go

```diff
@@ -459,30 +462,33 @@ func (s *service) createVM(requestCtx context.Context, request *proto.CreateVMRe
 	}()
 
 	s.logger.Info("creating new VM")
+	s.jailer, err = newJailer(requestCtx, s.logger, string(s.shimDir), s, request)
+	if err != nil {
+		return errors.Wrap(err, "failed to create jailer")
+	}
 
 	s.machineConfig, err = s.buildVMConfiguration(request)
 	if err != nil {
 		return errors.Wrapf(err, "failed to build VM configuration")
 	}
 
-	relSockPath, err := s.shimDir.FirecrackerSockRelPath()
-	if err != nil {
-		return errors.Wrapf(err, "failed to get relative path to firecracker api socket")
+	opts := []firecracker.Opt{
+		firecracker.WithLogger(s.logger),
 	}
 
-	relVSockPath, err := s.shimDir.FirecrackerVSockRelPath()
+	relVSockPath, err := s.jailer.JailPath().FirecrackerVSockRelPath()
 	if err != nil {
 		return errors.Wrapf(err, "failed to get relative path to firecracker vsock")
 	}
 
-	cmd := firecracker.VMCommandBuilder{}.
-		WithBin(s.config.FirecrackerBinaryPath).
-		WithSocketPath(relSockPath).
-		Build(s.shimCtx) // shimCtx so the VM process is only killed when the shim shuts down
+	jailedOpts, err := s.jailer.BuildJailedMachine(s.config, s.machineConfig, s.vmID)
+	if err != nil {
+		return errors.Wrap(err, "failed to build jailed machine options")
+	}
+	opts = append(opts, jailedOpts...)
 
 	// use shimCtx so the VM is killed when the shim shuts down
```

Comment:
I don't think this comment is accurate now because the default command builder will get overridden by our opts: https://github.com/firecracker-microvm/firecracker-go-sdk/blob/437f90243b6d40db06333829a6dc2c38e7570af4/machine.go#L266-L273

It doesn't hurt to still provide s.shimCtx but the comment should be removed.

Reply:
Cleared up the documentation on what occurs with the shim context.